File: Winter2022/examples/java/Lecture 7/RoomPersistenceLibraryDemo/app/src/main/java/mobiledev/unb/ca/roompersistencetest/entity/Item.java
package mobiledev.unb.ca.roompersistencetest.entity;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "item_table")  // Represents a SQLite table
public class Item {
    @PrimaryKey(autoGenerate = true)
    private int id;

    private String name;
    private int num;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getNum() {
        return num;
    }

    public void setNum(int num) {
        this.num = num;
    }
}

File: Fall2024/examples/Lecture 11/RoomPersistenceLibraryDemo/app/src/main/java/mobiledev/unb/ca/roompersistencetest/entity/Item.kt
package mobiledev.unb.ca.roompersistencetest.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "item_table") // Represents a SQLite table
class Item {
    @PrimaryKey(autoGenerate = true)
    var id = 0
    var name: String? = null
    var num = 0
}